CompositionRoot/Factories/AFactory.swift

```diff
@@ -0,0 +1,19 @@
+protocol AFactoryProtocol {
+    func createA() -> AProtocol
+}
+
+class AFactory: AFactoryProtocol {
+    private let applicationContext: ApplicationContext
+    
+    init(applicationContext: ApplicationContext) {
```

Comment:
The job of passing through ApplicationContext is moved from the original Business classes to Factory classes.  Doing this job make Factory classes depending on unnecessary dependencies, too.  Originally, Business classes depend on unnecessary dependencies. So what's the benefit by moving unnecessary dependencies from Business classes to Factory classes.
Factory classes are usually much simpler than Business classes.